### PR TITLE
Changes to support Modular Resources

### PR DESCRIPTION
**Pull request by Otterverse**
This mostly adds a new "module" API, and also adds IsMoving() to movable component types. As with the similar PR in goutils, this shouldn't merge until the main RDK mod resources PR is ready to merge, in case further changes are needed here.

### CODE REVIEW

File: proto/viam/module/v1/module.proto [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+package viam.module.v1;
+
+import "app/v1/robot.proto";
+import "robot/v1/robot.proto";
+
+option go_package = "go.viam.com/api/module/v1";
+
+// ModuleService deals with modular resource registration
+service ModuleService {
+  // AddResource tells a module about a new component to handle
```
Comment (cheukt) [inline proto/viam/module/v1/module.proto:12]: component -> resource

Comment (edaniels) [inline proto/viam/module/v1/module.proto:12]: component -> resource

**Review by edaniels — APPROVED** [review]
LGTM mod comments

File: proto/viam/app/v1/robot.proto [hunk 2]
```diff
@@ -69,6 +70,7 @@ message ComponentConfig {
   repeated string depends_on = 6;
   repeated ResourceLevelServiceConfig service_configs = 7 [(tagger.v1.tags) = "json:\"service_config\""];
   google.protobuf.Struct attributes = 8;
+  string api = 9;
```
[the remark this answers is not in the document]
Comment (Otterverse) [reply proto/viam/app/v1/robot.proto:73]: Yes, as it allows me to transmit only ComponentConfig as part of add resource. Without it, I can't tell if it's a service or a component on the other end, and would have to split AddComponent and AddService.

**Pull request merged**